lib/jekyll/gzip/compressor.rb

```diff
@@ -95,16 +95,21 @@ def self.compress_file(file_name, extensions: [], replace_file: false)
 
       private
 
+      def self.config(site)
+        config = site.config['gzip'] || {}
+        Jekyll::Gzip::DEFAULT_CONFIG.merge(config)
```

Comment:
Maybe you don't need to merge here, since you're picking default values anyway?  So the whole method could be `site.config['gzip'] || {}`

Reply:
Agreed!